misk/src/main/kotlin/misk/jdbc/Cluster.kt

```diff
@@ -0,0 +1,17 @@
+package misk.jdbc
+
+/**
+ * [Cluster] provides a consistent abstraction for
+ * interacting with various database-related resources [T]
+ * based on their role in the underlying database topology.
+ *
+ * Database clusters are often composed of a Read/Write master
+ * instance and a set of Read-Only replica instances.
+ *
+ * Examples include a Read-Only [DataSource] or a Read/Write
+ * [SessionFactory] in Hibernate.
+ */
+interface Cluster<T> {
+  fun writer() : T
+  fun reader() : T
```

Comment:
should this be `T?`?

Reply:
we should make that clear in the javadoc. I would find that somewhat surprising behavior.

Reply:
One thing to consider: if you don't provide a specific `reader` then you'll likely use `@Inject @Exemplar DataSource` directly rather than `@Inject @Exemplar DataSourceCluster` anyways. This saves having to null-check the `reader`.